rpc/send.go

```diff
@@ -216,9 +216,10 @@ func Send(opts Options, method string, addrs []net.Addr, getArgs func(addr net.A
 			}
 
 			errors++
+
 			// Since we have a reconnecting client here, disconnect errors are retryable.
-			if retryErr, ok := err.(retry.Retryable); err == io.ErrUnexpectedEOF ||
-				(ok && retryErr.CanRetry()) {
+			disconnected := err.Error() == "connection is shut down" || err == io.ErrUnexpectedEOF
```

Comment:
Looks like it might be! I'll see about adding this to my other (still open)
PR.
On Jul 27, 2015 8:01 PM, "Ben Darnell" <notifications@github.com> wrote:

> In rpc/send.go
> <https://github.com/cockroachdb/cockroach/pull/1824#discussion_r35602185>:
>
> >  			// Since we have a reconnecting client here, disconnect errors are retryable.
> > -			if retryErr, ok := err.(retry.Retryable); err == io.ErrUnexpectedEOF ||
> > -				(ok && retryErr.CanRetry()) {
> > +			disconnected := err.Error() == "connection is shut down" || err == io.ErrUnexpectedEOF
>
> Is this "connection is shut down" the same as == rpc.ErrShutdown (where
> rpc == net/rpc) or is it coming from somewhere else?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/1824/files#r35602185>.
>